stories/icons/mockIcon.stories.tsx

```diff
@@ -1,28 +0,0 @@
-import React from 'react';
-import { IconsProps, defaultProps, propTypes } from 'src/utils/icons';
-
-/**
- * We construct a mock Icon components to use Storybook's ArgsTable functionality.
```

Comment:
With our new `Icon` component we no longer need to do this, so I moved all the Icon logic into `index.stories.tsx` instead of having the separate files + MDX.